policy_engine: rename inbound/outbound to new phase names (#955)
ngrok/ngrok-docs: docs/http/traffic-policy/index.mdx

## Overview

> "This module enables you to assign a policy to your endpoints by defining a set of rules for three traffic management phases:
> `on_tcp_connect`, `on_http_request`, and `on_http_response`. These rules allow you influence and control traffic to and from
> your upstream service."

Reviewer: ```suggestion
This module enables you to assign a policy to your endpoints by defining a set of rules for three traffic management phases:
`on_tcp_connect`, `on_http_request`, and `on_http_response`. These rules allow you influence and control traffic to and from
your upstream service.
```
?

## Overview

This module allows you to assign a policy to your endpoints containing a series

> of `on_tcp_connect`, on_http_request`, and `on_http_response` rules that can be used to influence and control traffic

Reviewer 2: Missing a backtick before `on_http_request`!

This would be a good place to use the phase terminology. Maybe something like ...

"This module enables you to assign a policy to your endpoints by defining a set of rules for three traffic management phases: `on_tcp_connect`, `on_http_request`, and `on_http_response`. These rules allow you influence and control traffic to and from your upstream service."

Author: I like that, done.

> to and from your upstream service.

Reviewer: Hmm might need some rewrite in here because HTTP endpoints also support the `on_tcp_connect` phase

## Behavior

> Traffic Policy rules are evaluated sequentially in the order they are configured with `on_tcp_connect` and `on_http_request` rules taking effect before the

Reviewer 2: We should be consistent with capitalization throughout the docs, e.g. "Traffic Policy" or "traffic policy" but not a combo of both.

Author: in a blurb where we have already referenced `Traffic Policy`, is it ok to use `Policy` from then on out? Sometimes it can be a bit clunk to use "TrafficPolicy" multiple times within a few sentences 🤷

Reviewer 2: I would lean towards "Traffic Policy" when referring to the module and use "policy" otherwise. This aligns with very first sentence of the doc: "This module enables you to assign a policy to your endpoints ..."

e.g 
"The Traffic Policy module is the best thing ever"
"Policy rules are evaluated sequentially"
"If no expressions are specified on a policy rule, chaos ensues"

## Behavior

Policy rules are evaluated sequentially in the order they are configured with `on_http_request` rules taking effect before the
upstream server is reached and `on_http_response` rules taking effect after the upstream server responds. Whether or not the configured
actions are performed is determined at runtime by the expressions.

> 

Reviewer: Need `on_tcp_connect` as well. Also in these files seems we may want to be calling Policy 'Traffic Policy' instead?

### Action Execution

> If a traffic policy's expressions are evaluated as a match against a connection, the policy's actions will be executed. If multiple actions are defined
> on a traffic policy, the actions will execute sequentially.

Reviewer: ```suggestion
If the expressions of a policy rule evaluate to true, the policy's actions will be executed. If multiple actions are defined
on a traffic policy, the actions will execute sequentially.
```

### Configuration

| Parameter            | Description                                                                                                            |
| -------------------- | ---------------------------------------------------------------------------------------------------------------------- |
| **on_http_request**  | A list of policy rules that will be applied to inbound http traffic in the order specified.                            |

> | **on_http_response** | A list of policy rules that will be applied to outbound http traffic in the order specified.                           |

Reviewer: Hmm idk how this table got this way to begin with but these seems confusing that we have the phases defined in the same table as the rule properties?

Author: we can remove it from the table if we want, i don't mind! @nijikokun thoughts?
